filament/src/PostProcessManager.cpp

```diff
@@ -1308,27 +1308,53 @@ FrameGraphId<FrameGraphTexture> PostProcessManager::bloomPass(FrameGraph& fg,
         });
     }
 
+    // TODO: this is a feature flag that should be passed in from above.
+    // Reading from one mip while writing to a different mip is covered in the OpenGL ES 3.0
+    // specification under the heading  "Feedback Loops Between Textures and the Framebuffer".
+    // Some OpenGL implementations do not permit this behavior (e.g. macOS Chrome + WebGL 2.0).
+#if defined(__EMSCRIPTEN__)
+    const bool disableFeedback = true;
+#else
+    const bool disableFeedback = false;
+#endif
+
     struct BloomPassData {
         FrameGraphId<FrameGraphTexture> in;
         FrameGraphId<FrameGraphTexture> out;
+        FrameGraphId<FrameGraphTexture> stage;
         FrameGraphRenderTargetHandle outRT[kMaxBloomLevels];
+        FrameGraphRenderTargetHandle stageRT[kMaxBloomLevels];
     };
 
     // downsample phase
     auto& bloomDownsamplePass = fg.addPass<BloomPassData>("Bloom Downsample",
             [&](FrameGraph::Builder& builder, auto& data) {
                 data.in = builder.sample(input);
-                data.out = builder.createTexture("Bloom Texture", {
+                data.out = builder.createTexture("Bloom Out Texture", {
                         .width = width,
                         .height = height,
                         .levels = bloomOptions.levels,
                         .format = outFormat
                 });
                 data.out = builder.write(builder.sample(data.out));
 
+                if (disableFeedback) {
+                    data.stage = builder.createTexture("Bloom Stage Texture", {
+                            .width = width,
+                            .height = height,
+                            .levels = bloomOptions.levels,
+                            .format = outFormat
+                    });
+                    data.stage = builder.write(builder.sample(data.stage));
+                } else {
+                    data.stage = builder.write(builder.sample(data.out));
```

Comment:
one thing that makes me nervous here is that `data.out` becomes invalid after calling `sample`. So it's dangerous to have it around. At least we need a comment, and maybe reset it to 0.

Reply:
The current code is trying to make it so that "stage" is just an alias for "out".  Would is make sense to add an alias method to the builder for this?

Another idea is leaving `bloomPass` exactly as it is, and instead simply create a brand new method called `bloomPingPong`.  There will be some code duplication but maybe it will be safer.

@pixelflinger WDYT?